### ISSUE TITLE
App stuck when double clicking on image

### ISSUE THREAD

**Issue by kaushalyacs** [jamesmontemagno/MediaPlugin#472]
Hi,

In iOS when I double click on the image to be selected image picker view closed and app stuck.
This happens only in iOS.

Version Number of Plugin: 3.1.1
Device Tested On: iOS 7
Simulator Tested On: iPhone X iOS 11.2
Version of VS: Professional 2017
Version of Xamarin: 4.7.10.38

Have anyone experience this before?

Thanks
Kaushalya

**Comment by diegolengler** [comment]
- Yes. In my case, picker is closing (or hiding) more pages than it should be.

- Are you using modal pages?

**Comment by kaushalyacs** [comment]
@diegolengler Yes in my case also picker closes more pages. Yes modal navigation used.

**Comment by MitchBomcanhao** [comment]
Getting a similar issue when quickly tapping two different images in the image picker page, using 
media plugin 4.0.1.5, 
iOS 12.3, 
Xamarin forms 4.1.0.555618

My app is brought back into view and the page I was in (a modal page) is incorrectly dismissed, and subsequent attempts to load any modal page will fail with this error message
```
Warning: Attempt to present <Xamarin_Forms_Platform_iOS_ModalWrapper: 0x119399860> on <Xamarin_Forms_Platform_iOS_NavigationRenderer: 0x11c48b800> whose view is not in the window hierarchy!
```
I've tried updating the Media plugin package to 4.3.1-beta and haven't been able to repro the issue yet, but I did get an exception on the first attempt of reproducing the above issue:
```
**System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'MediaPickerController'.**
```

**Comment by MitchBomcanhao** [comment]
Using the latest beta plugin doesn't appear to prevent this issue from happening as I've seen this again.

**Comment by MitchBomcanhao** [comment]
for reference, having switched over to xamarin essentials, we still suffer from similar problems and we have to include the same workaround (popping all modal pages in the stack) in order for the application to continue functioning (without the workaround we effectively get a non-responding application where buttons appear unresponsive because the modal stack still contains the pages that got unexpectedly dismissed. :/
